runtime/compiler/env/j9methodServer.cpp

```diff
@@ -27,6 +27,66 @@
 #include "control/MethodToBeCompiled.hpp"
 #include "control/JITaaSCompilationThread.hpp"
 #include "exceptions/DataCacheError.hpp"
+#include "env/VMAccessCriticalSection.hpp"
+
+static inline UDATA getFieldType(J9ROMConstantPoolItem * cp, I_32 cpIndex)
+   {
+   J9ROMFieldRef * ref = (J9ROMFieldRef *) (&cp[cpIndex]);
+   J9ROMNameAndSignature * nameAndSignature = J9ROMFIELDREF_NAMEANDSIGNATURE(ref);
+   J9UTF8 * data = J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSignature);
+   UDATA j9FieldType = (UDATA)(J9UTF8_DATA(data)[0]);
+
+   switch (j9FieldType)
+      {    // cannot use characters because on s390 the compiler will generate an EBCDIC value
+           // internal JVM representation is in ASCII
+      case J9_SIGBYTE_BYTE:// byte 'B'
+        return (j9FieldType |= J9FieldTypeByte);
+      case J9_SIGBYTE_CHAR:// char 'C'
+        return (j9FieldType |= 0);
+      case J9_SIGBYTE_DOUBLE:// double 'D'
+        return (j9FieldType |= (J9FieldTypeDouble | J9FieldSizeDouble ));
+      case J9_SIGBYTE_FLOAT:// float 'F'
+        return (j9FieldType |= J9FieldTypeFloat);
+      case J9_SIGBYTE_INT:// int 'I'
+        return (j9FieldType |= J9FieldTypeInt);
+      case J9_SIGBYTE_LONG:// long 'J'
+        return (j9FieldType |=  (J9FieldTypeLong | J9FieldSizeDouble ));
+      case J9_SIGBYTE_SHORT:// short 'S'
+        return (j9FieldType |= J9FieldTypeShort);
+      case J9_SIGBYTE_BOOLEAN:// boolean 'Z'
+        return (j9FieldType |= J9FieldTypeBoolean);
+      default:// object
+        return (j9FieldType |= J9FieldFlagObject);
+      }
+
+   }
+
+// takes a J9FieldType and returns the DataTypes associated
+// along with the size.
+//
+static TR::DataType decodeType(U_32 inType)
```

Comment:
decodeType and getfieldType are both define in the CPP file. using them means we need to move to Header file. Since it is being used only here I don't think its is of use to move to header file.

Reply:
If we remove the static qualifier the linker should do the job.